### ISSUE TITLE
Memory leaks reported by cppcheck

### ISSUE THREAD

**Issue by fuchsto** [dash-project/dash#178]
~~~
$ cppcheck dart-impl/mpi
# --- snip ---
[dart-impl/mpi/src/dart_globmem.c:173]: (error) Memory leak: disp_set
[dart-impl/mpi/src/dart_globmem.c:424]: (error) Memory leak: disp_set
[dart-impl/mpi/src/dart_globmem.c:499]: (error) Memory leak: disp_set
[dart-impl/mpi/src/dart_globmem.c:249]: (error) Memory leak: disp_set
[dart-impl/mpi/src/dart_globmem.c:283]: (error) Memory leak: baseptr_set
# --- snip ---
[dart-impl/mpi/src/dart_team_group.c:154]: (error) Memory leak: res
[dart-impl/mpi/src/dart_team_group.c:509]: (error) Memory leak: res
~~~

**Comment by devreal** [comment]
I will have a look at this.

**Comment by devreal** [comment]
Ahh sorry, seems to have been fixed in #179 already. Thanks for the fix!

**Comment by fuchsto** [comment]
Ah, sorry, forgot to add a reference to this issue in the pull request.
